### PR TITLE
reduce two array sizes

### PR DESCRIPTION
**Pull request by neheb**
Fixes several -Wformat-truncation warnings.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by nikias** [comment]
Okay, let's change the filename/path buffer sizes to PATH_MAX and the truncated ones to PATH_MAX-x where x has the appropriate value, and make sure to also use snprintf?

**Comment by neheb** [comment]
That's...a bit more work but sure.

**Comment by neheb** [comment]
The previous commits introduced several errors on 64-bit systems. Added that as well.

File: src/ipsw.c [hunk 1]
```diff
@@ -704,14 +704,14 @@ int ipsw_download_fw(const char *fwurl, unsigned char* isha1, const char* todir,
 	}
 	fwfn++;
 
-	char fwlfn[256];
+	char fwlfn[PATH_MAX];
```
Comment (nikias) [inline src/ipsw.c:707]: I think this should be PATH_MAX-5

File: src/ipsw.c [hunk 2]
```diff
@@ -704,14 +704,14 @@ int ipsw_download_fw(const char *fwurl, unsigned char* isha1, const char* todir,
...
 	if (todir) {
 		sprintf(fwlfn, "%s/%s", todir, fwfn);
 	} else {
 		sprintf(fwlfn, "%s", fwfn);
 	}
 
-	char fwlock[256];
+	char fwlock[PATH_MAX + 5];
```
Comment (nikias) [inline src/ipsw.c:714]: ... and this one PATH_MAX, no?

**Comment by neheb** [comment]
Sure. Changed.

**Pull request merged**